src/Carnac.Logic/MessageController.cs

```diff
@@ -26,9 +28,13 @@ public KeysController(ObservableCollection<Message> keys, IMessageProvider messa
 
         public void Start()
         {
-            var messageStream = messageProvider.GetMessageStream(keyProvider.GetKeyStream()).Publish().RefCount();
+            var keyStream = keyProvider.GetKeyStream();
+            var messageStream = messageProvider.GetMessageStream(keyStream).Publish();
```

Comment:
I was thinking that maybe each of these things could become operators which is why I started removing the DI. Unsure if that is the way to go or not.